lib/Dialect/FIRRTL/Transforms/IMConstProp.cpp

```diff
@@ -392,15 +412,29 @@ LatticeValue IMConstPropPass::getExtendedLatticeValue(FieldRef value,
 
   auto constant = result.getConstant();
 
-  // If this is a BoolAttr then we are dealing with a special constant.
-  if (auto boolAttr = dyn_cast<BoolAttr>(constant)) {
-    // No extOrTrunc necessary for clock or reset types.
+  // No extOrTrunc necessary for bools or strings.
+  if (auto boolAttr = dyn_cast<BoolAttr>(constant))
     return LatticeValue(boolAttr);
-  }
+  if (auto strAttr = dyn_cast<StringAttr>(constant))
+    return LatticeValue(strAttr);
```

Comment:
maybe we can just return `result` for property type? `if(isa<PropertyType>()) return result;`

Reply:
Oh! I somehow missed I could just return `result`, hence handling the supported types explicitly re:constructing LatticeValue for the right thing. Good call, thanks!